lib/mayaHydra/hydraExtensions/adapters/materialNetworkConverter.cpp

```diff
@@ -484,6 +516,54 @@ class MayaHydraTransmissionMaterialAttrConverter : public MayaHydraComputedMater
     }
 };
 
+class MayaHydraOpenPBRTransmissionMaterialAttrConverter : public MayaHydraComputedMaterialAttrConverter
+{
+public:
+    SdfValueTypeName GetType() override { return SdfValueTypeNames->Float; }
+
+    VtValue GetValue(
+        MFnDependencyNode&      node,
+        const TfToken&          paramName,
+        const SdfValueTypeName& type,
+        const VtValue*          fallback = nullptr,
+        MPlugArray*             outPlug = nullptr) override
+    {
+        VtValue transmission = MayaHydraMaterialNetworkConverter::ConvertMayaAttrToValue(
+            node, "transmissionWeight", type, nullptr, outPlug);
+        // Combine transmission and Geometry Opacity
+        VtValue geometryOpacity = MayaHydraMaterialNetworkConverter::ConvertMayaAttrToValue(
+            node, "geometryOpacity", type, nullptr, outPlug);
+
+        if (!transmission.IsHolding<float>()) {
+            if (fallback) {
+                return *fallback;
+            }
+            TF_DEBUG(MAYAHYDRALIB_ADAPTER_GET)
+                .Msg("MayaHydraOpenPBRTransmissionMaterialAttrConverter::GetValue(): "
+                     "No float plug found with name: transmission and no "
+                     "fallback given");
+            return VtValue();
+        }
+
+        float val = 1.0f - transmission.UncheckedGet<float>();
+        if (val < 1.0e-4f) {
+            // Clamp lower value as an opacity of 0.0 in hydra makes the object fully transparent,
+            // but in VP2 we still see the specular highlight if any, avoiding 0.0 leads to the same
+            // effect in hydra.
+            val = 1.0e-4f;
+        }
+
+        float fGeometryOpacity = 1.0f;
+        if (geometryOpacity.IsHolding<float>()) {
+            fGeometryOpacity = geometryOpacity.UncheckedGet<float>();
+        }
+
+        val *= fGeometryOpacity;
```

Comment:
I think you could move this to line after 558 in the if block.

Reply:
yes, we can.

Reply:
Could you please combine line 560 with line 557 ?